client/opentracing/src/main/java/io/opentracing/APMSpan.java

```diff
@@ -84,6 +77,104 @@ protected void init(APMSpanBuilder builder, TraceReporter reporter) {
         traceContext.startProcessingNode();
     }
 
+    protected void initReferences(APMSpanBuilder builder, TraceReporter reporter) {
+        // Find primary reference
+        Reference primaryRef = findPrimaryReference(builder.references);
+
+        if (primaryRef != null) {
+            // Primary reference found, so it will provide the main 'parent' relationship
+            // to the existing trace instance. Other relationships will be recorded as
+            // correlation identifiers.
+            if (primaryRef.getReferenceType() == References.CHILD_OF) {
+                initChildOf(builder, reporter, primaryRef);
+    
+            } else if (primaryRef.getReferenceType() == References.FOLLOWS_FROM) {
+                initFollowsFrom(builder, reporter, primaryRef);
+            }
+        } else {
+            // No primary reference found, so means that all references will be treated
+            // as equal, to provide a join construct within a separate fragment.
+            initTopLevelState(this, reporter);
+
+            List<String> traceIds = builder.references.stream().map(ref -> {
```

Comment:
To be clarified - but even if that was supposed to be the case, we would need to flag this as an error, so would need to know that the application had passed spans associated with different traces. Will deal with once https://issues.jboss.org/browse/HWKAPM-753 is resolved.